ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerSubscription.scala

```diff
@@ -83,7 +87,10 @@ private[platform] case class ParallelIndexerSubscription[DB_BATCH](
           .buffered(metrics.daml.parallelIndexer.inputBufferLength, maxInputBufferSize)
       )
         .map(batch => batch.offsetsUpdates -> batch.lastSeqEventId)
-        // TODO LLP: Introduce mechanism (buffer) to allow observing downstream backpressure
+        .buffered(
```

Comment:
Is it intentional that we are buffering a batch of batches?
I am a little bit worries about the batchsize * maxTailerBatchSize * maxOutputBatchedBufferSize equation...
How about: 
* we change this to maxOutputBatchSize instead
* we change batching so it is batchWeighted, and weight would be the batch's size, and maxweight would be maxOutputBatchSize
* we remove buffering
* our signal metric for downstream backpressure would be simply the total weight of one weighted batch

Reply:
> Is it intentional that we are buffering a batch of batches?

Yes, but I look your idea with batchWeighted better 👍

Reply:
Discussed offline that it is fine as proposed.